File: backend/src/main/java/com/davnig/invoicemanagementapi/model/entity/ProfitabilityCoefficient.java
package com.davnig.invoicemanagementapi.model.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "t_profitability_coefficient")
@Getter
@Setter
public class ProfitabilityCoefficient {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "profitability-coeff-id-gen")
    @SequenceGenerator(name = "profitability-coeff-id-gen", sequenceName = "t_profitability_coefficient_id_seq", allocationSize = 1)
    private int id;
    private float coefficient;
    private String description;

}

File: backend/src/main/kotlin/com/davnig/invoicemanagementapi/model/entity/ProfitabilityCoefficient.kt
package com.davnig.invoicemanagementapi.model.entity

import jakarta.persistence.*

@Entity
@Table(name = "t_profitability_coefficient")
class ProfitabilityCoefficient {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "profitability-coeff-id-gen")
    @SequenceGenerator(
        name = "profitability-coeff-id-gen",
        sequenceName = "t_profitability_coefficient_id_seq",
        allocationSize = 1
    )
    var id = 0
    var coefficient = 0f
    var description: String? = null

}